### PR TITLE
Improve codespace secrets approach

### PR DESCRIPTION
**Pull request by ajparsons**
Features like the donate page and election lookup need external API keys, which doesn't work nicely with the docker setup because the conf/general is recreated each time the docker is opened (rather than created). 

This PR:

- Adjusts docker setup so that system environmental variables populate the `conf/general` file.
- Adjust 'devcontainer.json' to prompt codespaces users to setup the (optional) keys in their user codespace secrets store.

image

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -24,6 +24,13 @@ services:
       TWFY_TEST_DB_USER: twfy
       TWFY_TEST_DB_PASS: password
       TWFY_VOTES_URL: ${TWFY_VOTES_URL}
+      MAPIT_URL: ${MAPIT_URL:-https://mapit.mysociety.org/}
+      MAPIT_API_KEY: ${MAPIT_API_KEY:-}
+      DEMOCRACYCLUB_TOKEN: ${DEMOCRACYCLUB_TOKEN:-}
+      RECAPACHA_SITE_KEY: ${RECAPACHA_SITE_KEY:-}
```
Comment (dracos) [inline docker-compose.yml:30]: Some spelling mistakes in here

**Review by dracos — CHANGES_REQUESTED** [review]
Some typos to fix I think is all

File: docker-compose.yml (hunk 1 above)
Comment (ajparsons) [reply docker-compose.yml:30]: Should be fixed!

Comment (dracos) [reply docker-compose.yml:30]: And still mispelled here

**Comment by ajparsons** [comment]
Sorry sorry - should all be fixed.

**Review by dracos — APPROVED** [review]
Thanks. Didn't want that to cause some confusion down the line somewhere, if unlikely :)

**Pull request merged**